objects/utils.py

```diff
@@ -31,7 +31,11 @@ async def flash_with_customizations(status: str, msg: str, template: str) -> str
 
 def get_safe_name(name: str) -> str:
     """Returns the safe version of a username."""
-    return name.lower().replace(' ', '_')
+    # Safe name should meet few criterias.
+    # - Whole name should be lower letters.
+    # - Space must be replaced with _
+    # - Whitespace from right side should be removed.
+    return name.lower().replace(' ', '_').rstrip()
```

Comment:
.rstrip() won't do anything here as the spaces will already be replaced with _. it should stay anyways ;P all spaces should be preserved, even on the ends

Reply:
right my bad we can just switch the order of functions, but rstrip is actually useful, i had few cases when people actually left space after typing their nickname therefore they couldn't login/register

Reply:
https://github.com/cmyui/gulag/blob/master/misc/utils.py#L129-L131

Reply:
yea but that's on them (and they'll be able to login by using the correct amount of spaces), with the current gulag spec people *should* be able to have spaces on the sides of their names, this would break it for the existing users.